### PR TITLE
Fixes #23034 - Show whether subs are upstream or not

### PR DESCRIPTION
**Pull request by jturel**
We need a way to differentiate in the new manage subs UI - so we can check the upstream field in the subscriptions index API

### CODE REVIEW

**Comment by jturel** [comment]
@waldenraines this is relevant to your UI work

@akofink @jlsherrill here's what we discussed on IRC

@JacobCallahan - this PR has some tests using our fixtures

File: app/models/katello/pool.rb [hunk 1]
```diff
@@ -63,6 +63,10 @@ def products
       self.subscription.products if self.subscription
     end
 
+    def upstream?
+      subscription.redhat?
```
Comment (akofink) [inline app/models/katello/pool.rb]: `subscription` can be nil for virt_who subscriptions.

Comment (JacobCallahan) [inline app/models/katello/pool.rb]: does subscription.redhat? return true if it is a red hat sub? If so, then wouldn't this function return true if it is a downstream subscription? Or am I mixing something up in ruby land?

Comment (JacobCallahan) [reply app/models/katello/pool.rb]: OR... always another or.. in this case does "upstream" refer to subscriptions that reside on the RH end? If that is the case, then what "stream" are custom subs referred to?

Comment (akofink) [reply app/models/katello/pool.rb]: 'upstream' here means that it comes from the RHSM portal. AKA, a Red Hat subscription, not a custom or virt_who subscription.

Comment (JacobCallahan) [reply app/models/katello/pool.rb]: ahhh ok. i was thinking in the context of our development where upstream != redhat. thanks @akofink!

**Comment by jturel** [comment]
@akofink nice catch - updated

**Comment by jturel** [comment]
[test katello]

**Pull request merged**